From 1a76f8fd0f353f968331b40ab4c86658372972d3 Mon Sep 17 00:00:00 2001
From: John Johansen <john.johansen@canonical.com>
Date: Fri, 18 Mar 2016 06:05:49 -0700
Subject: [PATCH 1617/1767] Revert "UBUNTU: SAUCE: apparmor: Fix:
 label_vec_merge insertion"

BugLink: http://bugs.launchpad.net/bugs/1379535

This reverts commit 4cce60ba994d9c9e58d731c14cb257ca2c4ddeb6.

Signed-off-by: Tim Gardner <tim.gardner@canonical.com>
---
 security/apparmor/label.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/security/apparmor/label.c b/security/apparmor/label.c
index 627109a..218d856 100644
--- a/security/apparmor/label.c
+++ b/security/apparmor/label.c
@@ -1160,11 +1160,11 @@ struct aa_label *aa_label_vec_merge(struct aa_profile **vec, int len,
 	if (!new)
 		return NULL;
 
+	write_lock_irqsave(&ls->lock, flags);
 	for (i = 0; i < len; i++) {
 		new->ent[i] = aa_get_profile(vec[i]);
+		label = __aa_label_insert(ls, new, false);
 	}
-	write_lock_irqsave(&ls->lock, flags);
-	label = __aa_label_insert(ls, new, false);
 	write_unlock_irqrestore(&ls->lock, flags);
 	if (label != new)
 		/* not fully constructed don't put */
-- 
1.9.3

